pmm-app/src/pmm-update-panel/module.ts

```diff
@@ -197,30 +210,30 @@ export class PanelCtrl extends MetricsPanelCtrl {
      * Send request for get info about update status
      */
     private getLog($scope, $http): void {
-        if (!$scope.logLocation.length) return;
-
+        if ($scope.updateCntErrors > 600) {
+            console.log('error: ', $scope.errorMessage, $scope.updateCntErrors);
+            $scope.updateFailed = true;
+            return;
+        }
+        if ($scope.isUpdated) {
+            $scope.canBeReloaded = true;
+            return;
+        }
         $http({
-            method: 'GET',
-            url: $scope.logLocation,
+            method: 'POST',
+            url:  PanelCtrl.API.UPDATE_STATUS,
+            data: {'auth_token': $scope.updateAuthToken, 'log_offset': $scope.updateLogOffset}
         }).then(response => {
-            $scope.output = response.data.detail;
-            if (response.data.title === PanelCtrl.PROCESS_STATUSES.IN_PROGRESS) window.setTimeout(this.getLog.bind(this, $scope, $http), 1000);
-
-            if (response.data.title === PanelCtrl.PROCESS_STATUSES.DONE) {
-                this.reset($scope);
-                $scope.version = $scope.errorMessage ? $scope.version : $scope.nextVersion;
-                $scope.currentReleaseDate = $scope.errorMessage ? $scope.currentReleaseDate : $scope.newReleaseDate;
-                $scope.isUpdated = true;
-                $scope.canBeReloaded = true;
-            }
-            if (response.data.title === PanelCtrl.PROCESS_STATUSES.FAILED) {
-                $scope.isChecked = true;
-                $scope.isUpdateAvailable = true;
-                $scope.canBeReloaded = true;
-                $scope.errorMessage = PanelCtrl.ERRORS.UPDATE;
-            }
-        }).catch(() => {
-            this.reset($scope);
+             const data = response.data;
+             $scope.isUpdated = 'done' in data ? data.done : false;
+             $scope.updateLogOffset = 'log_offset' in data ? data.log_offset : 0;
+             $scope.output += '\n' + data.log_lines.join('\n');
+             window.setTimeout(this.getLog.bind(this, $scope, $http), 500);
+        }).catch((err) => {
+             $scope.updateCntErrors++;
```

Comment:
> Reset error counter when a non-error response is received.

Reply:
Good catch - removed it for debug and forgot to add back.